### ISSUE TITLE
Pay for order doesn't work with saved credit cards

### ISSUE THREAD

**Issue by ricardo** [woocommerce/woocommerce-gateway-stripe#2418]
**Describe the bug**

An error is thrown if you try to use a saved credit card in the pay for order page:

> `Undefined property: stdClass::$object in woocommerce-gateway-stripe/includes/abstracts/abstract-wc-stripe-payment-gateway.php on line 1451`

**To Reproduce**
1. Make sure you're logged in and have a credit added to your account. Also, try with a 3DS card.
2. Go to wp-admin > WooCommerce > Orders > Add order.
3. Create an order with any simple product item and then go to the "Customer payment page".
4. The error `Undefined property: stdClass::$object` is thrown in the logs. Checkout doesn't work.

**Additional context**
The fix may be different for saved 3DS cards like `4000000000003220`.

**Comment by Mayisha** [comment]
I tried the following steps locally with both `New checkout experience` enabled and disabled-

1. Logged in with an admin user.
2. Added credit cards to `my account` page (both normal and 3DS cards).
3. Went to `wp-admin > WooCommerce > Orders` page.
4. Clicked `Add order` and created an order with a simple product.
5. On the order page clicked `Customer payment page` and was taken to the pay for order page.
6. Completed payment with the normal saved card without any errors.
7. Repeated steps 3-5 and completed the order with the 3DS card without any error.

@ricardo is this issue still reproducible on your end?
